### PR TITLE
[DEV-1755] Add wallet connect

### PR DESCRIPTION
**Pull request by Kolibri1990**

### CODE REVIEW

**Review by TaprootFreak — APPROVED** [review]

File: src/components/home/connect-hint.tsx [hunk 1]
```diff
@@ -24,10 +24,13 @@ export function ConnectHint(props: ConnectHintProps): JSX.Element {
...
     case WalletType.META_MASK:
     case WalletType.ALBY:
+    case WalletType.WALLET_CONNECT:
       const confirmMessage =
         type === WalletType.META_MASK
           ? 'Please confirm the connection in your MetaMask.'
-          : 'Please confirm the connection in the Alby browser extension.';
+          : type === WalletType.ALBY
+          ? 'Please confirm the connection in the Alby browser extension.'
+          : 'Please confirm the connection in your wallet.';
```
Comment (davidleomay) [inline src/components/home/connect-hint.tsx:33]: Add translation to `de.json`

File: src/hooks/blockchain.hook.ts [hunk 2]
```diff
@@ -37,16 +38,22 @@ export function useBlockchain(): BlockchainInterface {
...
 
     const id = Object.entries(chainIds).find(([_, b]) => b === blockchain)?.[0];
     return id && web3.utils.toHex(id);
   }
 
+  function toChainId(blockchain: Blockchain): string | undefined {
+    const web3 = new Web3(Web3.givenProvider);
+    const id = Object.entries(chainIds).find(([_, b]) => b === blockchain)?.[0];
+    return id;
+  }
```
Comment (davidleomay) [inline src/hooks/blockchain.hook.ts:52]: Reuse code:

```ts
function toChainHex(blockchain: Blockchain): string | undefined {
    const web3 = new Web3(Web3.givenProvider);

    const id = toChainId(blockchain);
    return id && web3.utils.toHex(id);
}

function toChainId(blockchain: Blockchain): string | undefined {
   return Object.entries(chainIds).find(([_, b]) => b === blockchain)?.[0];
}
```

File: src/hooks/wallets/wallet-connect.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    const provider = await EthereumProvider.init({
+      projectId: process.env.REACT_APP_WC_PID ?? '40d5335cbc27edf36ef95389c9d1ac22',
+      chains: [+(chainId ?? 1)],
+      showQrModal: true,
+      metadata: {
+        name: 'DFX.swiss',
+        description: 'DFX.swiss',
+        url: 'https://dfx.swiss/',
+        icons: ['https://dfx.swiss/wp-content/uploads/2023/08/dfx-logo-neu.png'],
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.ts:35]: Use this icon: https://content.dfx.swiss/img/v1/website/logo-dark.svg

File: src/hooks/wallets/wallet-connect.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    const result = await client.request<string[]>({ method: 'eth_requestAccounts' });
+    return result[0];
+  }
+
+  async function setupConnection(blockchain: Blockchain): Promise<EthClient> {
+    const chainId = toChainId(blockchain);
+
+    const provider = await EthereumProvider.init({
+      projectId: process.env.REACT_APP_WC_PID ?? '40d5335cbc27edf36ef95389c9d1ac22',
+      chains: [+(chainId ?? 1)],
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.ts:29]: I would throw an error, if `process.env.REACT_APP_WC_PID` is not defined, and remove the fallback value.

File: src/hooks/wallets/wallet-connect.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    });
+    setWalletConnectClient(provider);
+    return provider;
+  }
+
+  async function signMessage(msg: string, address: string, blockchain: Blockchain): Promise<string> {
+    try {
+      const client = walletConnectClient ?? (await setupConnection(blockchain));
+      const signature = await client.request<string>({ method: 'personal_sign', params: [address, msg] });
+      return signature;
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.ts:46]:
```ts
      const client = walletConnectClient ?? (await setupConnection(blockchain));
      return await client.request<string>({ method: 'personal_sign', params: [address, msg] });
```

File: src/hooks/wallets/wallet-connect.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    const chainId = toChainId(blockchain);
+
+    const provider = await EthereumProvider.init({
+      projectId: process.env.REACT_APP_WC_PID ?? '40d5335cbc27edf36ef95389c9d1ac22',
+      chains: [+(chainId ?? 1)],
+      showQrModal: true,
+      metadata: {
+        name: 'DFX.swiss',
+        description: 'DFX.swiss',
+        url: 'https://dfx.swiss/',
```
[the remark this answers is not in the document]
Comment (davidleomay) [reply src/hooks/wallets/wallet-connect.ts:34]: Or use URL and title of current site?

File: src/hooks/wallets/wallet-connect.ts [hunk 7]
```diff
@@ -18,32 +19,35 @@ export function useWalletConnect(): WalletConnectInterface {
     const client = walletConnectClient ?? (await setupConnection(blockchain));
     await client.connect();
     const result = await client.request<string[]>({ method: 'eth_requestAccounts' });
+    console.log(result);
     return result[0];
```
Comment (davidleomay) [inline src/hooks/wallets/wallet-connect.ts]: Remove

**Comment by TaprootFreak** [comment]
faild

**Review by davidleomay — APPROVED** [review]

**Pull request merged**